src/run_docker.py

```diff
@@ -155,6 +156,9 @@ async def run_single_polling_with_timeout(services):
         if config.keep_refreshing_extensions_config():
             new_services_from_extensions_task = asyncio.create_task(prepare_services_config_for_next_polling(services))
 
+        if config.metric_autodiscovery():
+            services = await prepare_services_autodiscovery_polling(services)
```

Comment:
I think name enrich_services_with_autodiscovery_metrics for this top-level function is much better
prepare services autodiscovery polling - this name is a little confusing I think